### PR TITLE
tests: moved few unittest tests to pytest

### PR DESCRIPTION
**Pull request by nik123**
Part of #1819 

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

PR doesn't require documentation updates.

### CODE REVIEW

File: tests/unit/test_prompt.py [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-    @mock.patch("sys.stdout.isatty", return_value=True)
-    @mock.patch("dvc.prompt.input", side_effect=EOFError)
-    def test_eof(self, mock_input, mock_isatty):
-        ret = confirm("message")
-        mock_isatty.assert_called()
-        mock_input.assert_called()
-        self.assertFalse(ret)
+def test_confirm_in_tty_if_stdin_is_closed():
+    with mock.patch("dvc.prompt.input", side_effect=EOFError) as mock_input:
+        with mock.patch("sys.stdout.isatty", return_value=True) as mock_isatty:
```
Comment (efiop) [inline tests/unit/test_prompt.py:8]: Let's use `mocker` fixture instead.

File: tests/unit/test_daemon.py [hunk 2]
```diff
@@ -1,34 +1,33 @@
...
-
-        file_path = os.path.abspath(inspect.stack()[0][1])
-        file_dir = os.path.dirname(file_path)
-        test_dir = os.path.dirname(file_dir)
-        dvc_dir = os.path.dirname(test_dir)
-        self.assertEqual(env["PYTHONPATH"], dvc_dir)
-        self.assertEqual(env[daemon.DVC_DAEMON], "1")
+def test_daemon():
+    with mock.patch("dvc.daemon._spawn_windows") as mock_windows:
+        with mock.patch("dvc.daemon._spawn_posix") as mock_posix:
```
Comment (efiop) [inline tests/unit/test_daemon.py:10]: Same here, we should use `mocker` fixture.

File: tests/unit/test_daemon.py [hunk 3]
```diff
@@ -1,38 +1,35 @@
...
+    file_dir = os.path.dirname(file_path)
+    test_dir = os.path.dirname(file_dir)
+    dvc_dir = os.path.dirname(test_dir)
+    assert env["PYTHONPATH"] == dvc_dir
+    assert env[daemon.DVC_DAEMON] == "1"
+
+
+def test_no_recursive_spawn(mocker):
+    mocker.patch.dict(os.environ, {daemon.DVC_DAEMON: "1"})
+    mock_spawn = mocker.patch("dvc.daemon._spawn")
```
Comment (skshetry) [inline tests/unit/test_daemon.py:34]: Call to the daemon is missing.
```suggestion
    mock_spawn = mocker.patch("dvc.daemon._spawn")
    daemon.daemon(["updater"])
```

File: tests/unit/test_prompt.py (hunk 1 above)
Comment (nik123) [reply tests/unit/test_prompt.py]: BTW it seems there are quite a few `unittest.mock` usages inside pytest tests. For example in https://github.com/iterative/dvc/blob/d8b67f6f78781154aab39783401c181089468b02/tests/func/test_analytics.py#L9-L14

Should all `unittest.mock` usages be eliminated as a part of #1819?

**Review by skshetry — APPROVED** [review]

**Comment by efiop** [comment]
Thank you @nik123 ! 🙏

**Pull request merged**